Map/MapBase.gd

```diff
@@ -0,0 +1,7 @@
+extends Resource
+class_name Map
+## Class for a map. Provides classes for the map
+##
+## Holds basic functionality for a map
+@export var floors: Array[Floor]
```

Comment:
This should be a 2D array of rooms since the floor implementation was decided against.